test/integration/authz_v2_test.go

```diff
@@ -0,0 +1,399 @@
+	applyServices(t, e.Services, serviceAction)
+	validateAuthorizationPolicy(t, e.AuthorizationPolicies)
+}
+
+// createServiceObjects - Creates the list of kubernetes service objects as passed in
+func createServiceObjects(t *testing.T, services []*v1.Service) {
+	for _, service := range services {
+		_, err := framework.Global.K8sClientset.CoreV1().Services(service.Namespace).Create(service)
+		assert.Nil(t, err, "service create error should be nil")
+	}
+}
+
+// updateServiceObjects - Updates the list of Kubernetes service objects as passed in
+func updateServiceObjects(t *testing.T, services []*v1.Service) {
+	for _, service := range services {
+		currentService, err := framework.Global.K8sClientset.CoreV1().Services(service.Namespace).Get(service.Name, metav1.GetOptions{})
+		assert.Nil(t, err, "service get error should be nil")
+		service.ResourceVersion = currentService.ResourceVersion
+		service.Spec.ClusterIP = currentService.Spec.ClusterIP
+		_, err = framework.Global.K8sClientset.CoreV1().Services(service.Namespace).Update(service)
+		assert.Nil(t, err, "service update error should be nil")
+	}
+}
+
+// deleteServiceObjects - Delete the list of kubernetes services passed in
+func deleteServiceObjects(t *testing.T, services []*v1.Service) {
+	for _, service := range services {
+		err := framework.Global.K8sClientset.CoreV1().Services(service.Namespace).Delete(service.Name, &metav1.DeleteOptions{})
+		assert.Nil(t, err, "service delete error should be nil")
+	}
+}
+
+// cleanupAuthorizationRbac Deletes the Athenz domains and deletes the services
+func cleanupAuthorizationRbac(t *testing.T, e *fixtures.ExpectedV2Rbac) {
+	applyAthenzDomain(t, e.AD, delete)
+	deleteServiceObjects(t, e.Services)
+}
+
+// applyAthenzDomain Based on the action specified - creates, updates or deletes athenzDomain
+func applyAthenzDomain(t *testing.T, athenzDomain *athenzdomain.AthenzDomain, a action) {
+	switch a {
+	case create:
+		_, err := framework.Global.AthenzDomainClientset.AthenzV1().AthenzDomains().Create(athenzDomain, metav1.CreateOptions{})
+		assert.Nil(t, err, "athenz domain create error should be nil")
+	case update:
+		currentAD, err := framework.Global.AthenzDomainClientset.AthenzV1().AthenzDomains().Get(athenzDomain.Name, metav1.GetOptions{})
+		assert.Nil(t, err, "athenz domain get error should be nil")
+		athenzDomain.ResourceVersion = currentAD.ResourceVersion
+		_, err = framework.Global.AthenzDomainClientset.AthenzV1().AthenzDomains().Update(athenzDomain, metav1.UpdateOptions{})
+		assert.Nil(t, err, "athenz domain update error should be nil")
+	case delete:
+		err := framework.Global.AthenzDomainClientset.AthenzV1().AthenzDomains().Delete(athenzDomain.Name, metav1.DeleteOptions{})
+		assert.Nil(t, err, "athenz domain delete error should be nil")
+	}
+}
+
+// applyServices Based on the action specified creates, updates or deletes services
+func applyServices(t *testing.T, services []*v1.Service, a action) {
+	switch a {
+	case create:
```

Comment:
can you make applyServices and applyAthenzDomain in the same format? Either have helper function call in each case or just include the code for each case

Reply:
Done